app/models/topic.rb

```diff
@@ -1102,18 +1110,25 @@ def make_banner!(user)
       .update_all(dismissed_banner_key: nil)
 
     self.archetype = Archetype.banner
+    self.bannered_until = bannered_until
     self.add_small_action(user, "banner.enabled")
     self.save
 
     MessageBus.publish('/site/banner', banner)
+
+    Jobs.cancel_scheduled_job(:remove_banner, topic_id: self.id)
+    Jobs.enqueue_at(bannered_until, :remove_banner, topic_id: self.id) if bannered_until
```

Comment:
```suggestion
    Jobs.enqueue_at(bannered_until, :remove_banner, topic_id: self.id) if bannered_until.present?
```

`bannered_until` defaults to a blank string unfortunately so we need a presence check.

Reply:
I was making changes to this while you were doing your comment, can you check again please?